samples/extensions/full_screen_exclusive/full_screen_exclusive.h

```diff
@@ -0,0 +1,134 @@
+/* Copyright (c) 2023, Holochip Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Demonstrates and showcases full_screen_exclusive and related functionalities
+ */
+
+#pragma once
+
+#include "common/vk_common.h"
+#include "core/instance.h"
+#include "platform/application.h"
+
+class FullScreenExclusive : public vkb::Application
```

Comment:
Is there a reason, you're deriving your sample from vkb::Application and not from ApiVulkanSample or at least vkb::VulkanSample?

Reply:
Ah yes. Reason for me to use vkb::Application as its base class is that, I can very easily get access to the swapchain, swapchain images, and image views (i.e., "swapchain-buffer"). The high-level access requires some "walk around" to access these variables (i.e., they were all defined inside "render-content" class). By directly using vkb::Application as a base class, it saves a lot of work, and it is more straightforward to showcase the fullscreen exclusive features.